token/src/instruction.rs

```diff
@@ -85,6 +85,17 @@ pub enum TokenInstruction {
     ///   2. `[writable]` Token being burned.
     ///   3. Optional: `[writable]` Source account if key 1 is a delegate account.
     Burn(u64),
+
+    /// Wrap SOL into a token account.  The full lamport balance in the new account will be wrapped
+    ///   0. `[writable]` New wrapped account being created.
+    ///   1. `[]` Owner of the new account.
+    CreateWrappedAccount,
+
+    /// Deallocate and release all native SOL held by a token account
+    ///   0. `[signer]` Owner of the token account.
+    ///   1. `[writable]` Token account to release.
+    ///   2. `[writable]` Destination account to receive the full SOL balance of the account.
+    ReleaseAccount,
```

Comment:
`ReleaseAccount` works for non-wrapped token accounts as well, and thus is effectively a replacement for `Burn(u64)` but also:
1) Prevents a zero-balance token account that floats around forever
2) Assumes that we'll redefine the `supply` field according to https://github.com/solana-labs/solana-program-library/issues/66#issuecomment-651928634 such that it never decreases

Reply:
If we don't have the requirement to partial burn tokens from an account then we should remove the `burn` instruction :-)

Reply:
Yeah, totally agree!   And you can still do a partial burn by transferring to a new token account then `ReleaseAccount`ing it.

Reply:
Though that is a two-step process (one transaction), will have a performance impact depending on how fast folks want to burn stuff

Reply:
I feel this is unlikely to be an issue, but let's see what comes out of that slack thread.

Reply:
Ok fine, no changes to `Burn` since partial burns seem to be a little popular.

For a normal non-wrapped token, we could:
1. Consider `ReleaseAccount` to also act as a full `Burn`
2. Require the non-wrapped token have a zero token balance before `ReleaseAccount` can be called on it
3. Only allow `ReleaseAccount` on wrapped tokens, which means a rent-exempt non-wrapped token account will always live forever.

I prefer 1.

Reply:
I prefer 1 also